File: Course_1-Algorithmic_Toolbox/Week-3/Excercise_Challenges/4_maximum_advertisement_revenue/dot_product.py
def max_dot_product(a, b):
    a.sort()
    b.sort()
    result = 0

    for i in range(len(a)):
        result += a[i] * b[i]

    return result

if __name__ == '__main__':
    n = int(input())
    a = list(map(int, input().split()))
    b = list(map(int, input().split()))
    print(max_dot_product(a, b))
    

File: Course_1-Algorithmic_Toolbox/Week-3/Excercise_Challenges/4_maximum_advertisement_revenue/dot_product.cpp
#include <algorithm>
#include <iostream>
#include <vector>
typedef long long ll;
using namespace std;

ll max_dot_product(vector<int> a, vector<int> b)
{
    ll result = 0;
    int n = a.size();
    sort(a.begin(), a.end());
    sort(b.begin(), b.end());
    
    for(int i = 0; i < n; ++i)
        result += (ll)a[i] * b[i];

    return result;
}

int main(void)
{
    int n;
    cin >> n;
    
    vector<int> a(n), b(n);
    for(int i = 0; i < n; ++i){
        cin >> a[i];
    }
    for(int i = 0; i < n; ++i){
        cin >> b[i];
    }

    cout << max_dot_product(a, b) << endl;
    
    return 0;
}
